### PR TITLE
chore: use eslint

### PR DESCRIPTION
**Pull request by paul-soporan**
The formatting in this repo is currently very inconsistent.

I've added `eslint` with `@yarnpkg/eslint-config`. Feel free to tweak things.

### CODE REVIEW

File: sources/demos/advanced.ts [hunk 1]
```diff
@@ -1,147 +1,142 @@
...
-    cwd: string;
-    stdin: Readable;
-    stdout: Writable;
-    stderr: Writable;
-}
-
-declare module 'yup' {
-    interface Schema<T> {
-        atMostOneOf(keys: string[]): this;
-    }
```
Comment (paul-soporan) [inline sources/demos/advanced.ts:16]: `yup` leftover

File: sources/advanced/builtins/index.ts [hunk 2]
```diff
@@ -1,3 +1,3 @@
-export * from './definitions'
-export * from './help'
-export * from './version'
\ No newline at end of file
+export * from './definitions';
+export * from './help';
+export * from './version';
```
Comment (merceyz) [inline sources/advanced/builtins/index.ts:3]: Does the eslint config not enforce a newline at the end of files? :thinking:

Comment (paul-soporan) [reply sources/advanced/builtins/index.ts:3]: Looks like it doesn't:
image

Comment (paul-soporan) [reply sources/advanced/builtins/index.ts:3]: https://github.com/yarnpkg/berry/pull/2287

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  extends: [
+    `@yarnpkg`,
+    `@yarnpkg/eslint-config/react`,
```
Comment (merceyz) [inline .eslintrc.js:4]:
```suggestion
```

Don't think this is needed as there is no react here

Comment (paul-soporan) [reply .eslintrc.js:4]: It's used inside the website (e.g. https://github.com/arcanis/clipanion/blob/master/website/src/pages/index.js), as Docusaurus uses React.

Comment (merceyz) [reply .eslintrc.js:4]: Ah, of course, never mind me :sweat_smile:

**Pull request merged**